src/shared/ptyfwd.c

```diff
@@ -1115,7 +1115,13 @@ int pty_forward_set_title(PTYForward *f, const char *title) {
         if (f->out_buffer_size > 0)
                 return -EBUSY;
 
-        return free_and_strdup(&f->title, title);
+        /* Truncate the title to 128 chars, since some terminal emulators really don't like overly long ANSI
+         * sequences */
+        _cleanup_free_ char *ellipsized = ellipsize(title, 128, 66);
```

Comment:
assert(title) or check NULL above? Otherwise the assertion in ellipsize_mem() could be triggered.